shared/gh/api/gh.api.util.js

```diff
@@ -47,23 +58,49 @@ define(['exports', 'moment', 'sinon', 'bootstrap-notify'], function(exports, mom
     };
 
     /**
-     * Get the date range the calendar should be displaying. The range starts on January 1st of the
-     * previous year and ends December 31st of the next year.
+     * Get the date range the calendar should be displaying. The date is determined by the calendar's current view.
+     *
+     *  * Day:
+     *  * We only fetch the events that occur on the specific day.
```

Comment:
This doesn't seem to be true. If I'd fetch at the time of writing it would get:

- everything starting from 15:51 on Wednesday the 21st
- everything today the 22nd
- everything tomorrow the 23rd until 15:51

The documentation suggest that we'd get a single full day (00:00 to 00:00) and I think it makes most sense to implement it like that. Especially since we're not actually using the data requested from 15:51 the previous day and tomorrow until 15:51 (that data gets requested again once we request those days). So, technically, every request we do for the calendar, we're requesting an extra 2 half days of events without ever using them.

The same principle applies for weeks and months. If we're going to request one month before and after the month that's being viewed we need to actually use that data or limit or request to the exact amount of data we need.